### DISCUSSION TITLE
Edit Fields Not working; not updating, and sending empty array to backend

### DISCUSSION THREAD

**Discussion by robstarkdev** [laravel/nova-issues#5122] [Q&A] [answered]
I am using Laravel version 9.41 and Nova 4. I can delete records, however it appears that on the frontend/JavaScript side of things my form fields are not having their contents captured and sent back to the server. My action_events table shows [ ] empty arrays as what is being sent.

I have spent a good four or five hours googling and checking that my Resources, field names etc. are all spelled correctly. I also see no error messages. Each time I get the green success message, "This has been updated!" or whatever.

`The [bounty](https://stackoverflow.com/help/bounty) expires in 6 days. Answers that other users post to this question are eligible for a +50 reputation bounty. [WebDev-SysAdmin](https://stackoverflow.com/users/5544593/webdev-sysadmin) wants to draw more attention to this question.
I am using Laravel version 9.41 and Nova 4. I can delete records, however it appears that on the frontend/JavaScript side of things my form fields are not having their contents captured and sent back to the server. My action_events table shows [ ] empty arrays as what is being sent.

I have spent a good four or five hours googling and checking that my Resources, field names etc. are all spelled correctly. I also see no error messages. Each time I get the green success message, "This has been updated!" or whatever.

use Illuminate\Http\Request;
use Laravel\Nova\Fields\ID;
use Laravel\Nova\Fields\Text;
use Laravel\Nova\Http\Requests\NovaRequest;

class Post extends Resource
{
    /**
     * The model the resource corresponds to.
     *
     * @var class-string<\App\Models\Post>
     */
    public static $model = \App\Models\Post::class;

    /**
     * The single value that should be used to represent the resource when being displayed.
     *
     * @var string
     */
    public static $title = 'id';

    /**
     * The columns that should be searched.
     *
     * @var array
     */
    public static $search = [
        'id',
        'text'
    ];

    /**
     * Get the fields displayed by the resource.
     *
     * @param  \Laravel\Nova\Http\Requests\NovaRequest  $request
     * @return array
     */
    public function fields(NovaRequest $request)
    {
        return [
            ID::make('id', 'id')->sortable(),
            Text::make('text', 'text')->sortable()
        ];
    }`

And here is my Model: 

`class Post extends Model
{
    use HasFactory;

    public $id;
    public $text;
    public $created_at;
    public $updated_at;

    protected $fillable = [
        'id',
        'text',
        'created_at',
        'updated_at'
    ];
}
`

**Comment by crynobone** [comment, marked as answer]
```php
class Post extends Model
{
    use HasFactory;

    public $id;
    public $text;
    public $created_at;
    public $updated_at;

    protected $fillable = [
        'id',
        'text',
        'created_at',
        'updated_at'
    ];
}
```

There no reason to define the following property. It's not part of any Eloquent conversion and the root cause of your issue:

```php
    public $id;
    public $text;
    public $created_at;
    public $updated_at;
```

**Reply by robstarkdev** [reply]
Thank you that solved it!!! But will I be able to use my Post model in other contexts as I normally would? ie using eloquent models etc.?

**Reply by robstarkdev** [reply]
Also, could you be so kind as to point me in the right direction to understanding "not part of any Eloquent conversion"? I thought we normally need to define those properties on our models. At any rate, it works now! thanks again!!
